controlpanel/api/cluster.py

```diff
@@ -346,3 +356,120 @@ def get_repository(user, repo_name):
         return github.get_repo(repo_name)
     except GithubException.UnknownObjectException:
         return None
+
+
+class ToolDeploymentError(Exception):
+    pass
+
+
+def deploy_tool(tool, user, **kwargs):
+    values = {
+        'username': user.username.lower(),
+        'Username': user.username.lower(),  # XXX backwards compatibility
+        'aws.iamRole': user.iam_role_name,
+        'toolsDomain': settings.TOOLS_DOMAIN,
+    }
+
+    # generate per-deployment secrets
+    for key, value in tool.values.items():
+        if value == '<SECRET_TOKEN>':
+            tool.values[key] = secrets.token_hex(32)
+
+    values.update(tool.values)
+    values.update(kwargs)
+    values = ','.join(f'{key}={val}' for key, val in values.items())
+
+    try:
+        return helm.upgrade_release(
+            f'{tool.chart_name}-{user.slug}',
+            f'mojanalytics/{tool.chart_name}',  # XXX assumes repo name
+            f'--version', tool.version,
+            f'--namespace', user.k8s_namespace,
+            f'--set', values,
+        )
+
+    except HelmError as error:
+        raise ToolDeploymentError(error)
+
+
+def list_tool_deployments(user, search_name=None, search_version=None):
+    deployments = []
+    k8s = KubernetesClient()
+    results = k8s.AppsV1Api.list_namespaced_deployment(user.k8s_namespace)
+    for deployment in results.items:
+        app_name = deployment.metadata.labels["app"]
+        _, version = deployment.metadata.labels["chart"].rsplit("-", 1)
+        if search_name and search_name not in app_name:
+            continue
+        if search_version and not version.startswith(search_version):
```

Comment:
Do we need to check the version? I'm wondering if this is causing the problem we were talking about [in the other comment](https://github.com/ministryofjustice/analytics-platform-control-panel/pull/732/files#r321617248) where if the user is using a different version is not shown/can't restart.

I've tried to change the version of the RStudio `Tool` record in the DB locally and the list of tools got confused/confusing. I couldn't see my RStudio instance and it seemed as if the RStudio helm chart wasn't installed and offered me to "Deploy" that version I didn't have.

I tried to comment this `if` statement and it doesn't seem to help 🤔. I think completely hide an instance if it's older than current version would make our user very confused and unhappy. I'll play with it locally in case I have some good idea ☺️